casanovo/denovo/model_runner.py

```
@@ -524,10 +525,14 @@ def initialize_model(self, train: bool, db_search: bool = False) -> None:
                 loaded_model_params.keys()
             )
             for param in architecture_params:
-                if model_params[param] != self.model.hparams[param]:
+                checkpoint_value = self.model.hparams.get(param)
+                if checkpoint_value is None:
+                    # New param not in old checkpoint; use config value.
+                    continue
+                if model_params[param] != checkpoint_value:
                     warnings.warn(
```

Ruff (0.15.6)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)